Formula/azure-cli.rb

```diff
@@ -1,36 +1,46 @@
-require "language/node"
-
 class AzureCli < Formula
-  desc "Official Azure CLI"
-  homepage "https://github.com/azure/azure-xplat-cli"
-  url "https://github.com/Azure/azure-xplat-cli/archive/v0.10.12-May2017.tar.gz"
-  version "0.10.13"
-  sha256 "60194e770b8dca0485db9d4c99f9cd432f7b43096cc5ad0353f52fd5b1b29181"
-  head "https://github.com/azure/azure-xplat-cli.git", :branch => "dev"
-
-  bottle do
-    cellar :any_skip_relocation
-    sha256 "0849685ed65f1bfa991c3ffbfaa281f9051d33b626b79b12b1f5341f5925976d" => :sierra
-    sha256 "6bcf4ea33dad69d8c703724e09468e08c98d053456fb41a94f6b5877d9a8c243" => :el_capitan
-    sha256 "16184e20ae089b157c2e513c1b194ace66d1f46da5eb2869fb9261b1f71aaf1f" => :yosemite
-  end
+  include Language::Python::Virtualenv
 
-  depends_on "node"
-  depends_on :python => :build
+  desc "Microsoft Azure CLI 2.0"
+  homepage "https://docs.microsoft.com/en-us/cli/azure/overview"
+  url "https://azurecliprod.blob.core.windows.net/releases/azure-cli_packaged_0.2.9.tar.gz"
+  version "2.0.2.9"
+  sha256 "6f949162d6bbf6848177cc009d60520c12cf2c1dfc068e9e5bbeac0ff6e55e4f"
+  head "https://github.com/Azure/azure-cli.git"
+
+  depends_on :python
 
   def install
-    rm_rf "bin/windows"
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
-    (bash_completion/"azure").write Utils.popen_read("#{bin}/azure --completion")
+    virtualenv_create(libexec)
+    bin_dir = libexec/"bin"
+
+    components = [
+      "src/azure-cli",
+      "src/azure-cli-core",
+      "src/azure-cli-nspkg",
+      "src/azure-cli-command_modules-nspkg",
+    ].map { |x| Pathname.new(x) } + Pathname.glob("src/command_modules/azure-cli-*/")
+
+    # Create wheel distribution of included components
+    components.each do |c|
+      c.cd { system bin_dir/"python", "setup.py", "bdist_wheel", "-d", buildpath/"wheel" }
+    end
+
+    # Install CLI from wheel distributions
```

Comment:
Actually we need to build these from source. CC @tdsmith am I missing something or is this building from binaries since `--no-binary all:` isn't being passed below and it's not using our `venv.pip_install`?